src/main/kotlin/io/github/inductiveautomation/kindling/core/Tool.kt

```diff
@@ -51,6 +51,8 @@ interface Tool : KindlingSerializable {
 
     val filter: FileFilter
```

Comment:
Minor: You can give this a default impl and save some duplication in downstream classes: `get() = FileFilter(description, *extensions)`